packages/server/src/course/course.service.ts

```diff
@@ -101,4 +114,117 @@ export class CourseService {
+      );
+      if (!sectionGroup)
+        throw new BadRequestException(
+          ERROR_MESSAGES.courseController.sectionGroupNotFound,
+        );
+      const khourySemesterParsed = this.loginCourseService.parseKhourySemester(
+        sectionGroup.semester,
+      );
+      const semester = await SemesterModel.findOne({
+        where: {
+          season: khourySemesterParsed.season,
+          year: khourySemesterParsed.year,
+        },
+      });
+      if (!semester)
+        throw new BadRequestException(
+          ERROR_MESSAGES.courseController.noSemesterFound,
+        );
+
+      // checks that course hasn't already been created
+      let course = await CourseModel.findOne({
+        where: {
+          name: courseParams.name,
+          semesterId: semester.id,
+        },
+      });
+      if (course)
+        throw new BadRequestException(
+          ERROR_MESSAGES.courseController.courseAlreadyRegistered,
+          courseParams.name,
+        );
+
+      try {
+        // create the submitted course
+        course = await CourseModel.create({
+          name: courseParams.name,
+          sectionGroupName: courseParams.sectionGroupName,
+          coordinator_email: courseParams.coordinator_email,
+          icalURL: courseParams.iCalURL,
+          semesterId: semester.id,
+          enabled: true,
+          timezone: courseParams.timezone,
+        }).save();
+      } catch (err) {
+        console.error(err);
+        throw new HttpException(
+          ERROR_MESSAGES.courseController.createCourse,
+          HttpStatus.INTERNAL_SERVER_ERROR,
+        );
+      }
+
+      try {
+        // create CourseSectionMappings for each crn
+        new Set(sectionGroup.crns).forEach(async (crn) => {
+          await CourseSectionMappingModel.create({
+            crn: crn,
+            courseId: course.id,
+          }).save();
+        });
+      } catch (err) {
+        console.error(err);
+        throw new HttpException(
+          ERROR_MESSAGES.courseController.createCourseMappings,
+          HttpStatus.INTERNAL_SERVER_ERROR,
+        );
+      }
+
+      // Add UserCourse to course
+      await UserCourseModel.create({
+        userId,
+        courseId: course.id,
+        role: Role.PROFESSOR,
+      }).save();
+
+      try {
+        // Update professor's last registered semester to semester model's current semester
+        let profLastRegistered: LastRegistrationModel;
+        profLastRegistered = await LastRegistrationModel.findOne({
+          where: { profId: userId },
+        });
+        if (profLastRegistered) {
+          profLastRegistered.lastRegisteredSemester = sectionGroup.semester;
+          await profLastRegistered.save();
+        } else {
+          profLastRegistered = await LastRegistrationModel.create({
+            profId: userId,
+            lastRegisteredSemester: sectionGroup.semester,
+          }).save();
+        }
+      } catch (err) {
+        console.error(err);
```

Comment:
I don't think you have to console.err here (since it will be caught in the controller and logged there)

Reply:
alternatively you could get rid of the console.err in the controller, since at that level you'll be catching your custom thrown HttpException -  I think it makes sense to log whatever low level typeORM error we get, then just throw the our custom HttpException.